specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/settings.json

```diff
@@ -0,0 +1,208 @@
+{
+    "swagger": "2.0",
+    "info": {
+      "title": "KeyVaultClient",
+      "description": "The key vault client performs cryptographic key operations and vault operations against the Key Vault service.",
+      "version": "7.4-preview.1"
+    },
+    "x-ms-parameterized-host": {
+      "hostTemplate": "{vaultBaseUrl}",
+      "useSchemePrefix": false,
+      "positionInOperation": "first",
+      "parameters": [
+        {
+          "name": "vaultBaseUrl",
+          "description": "The vault name, for example https://myvault.vault.azure.net.",
+          "required": true,
+          "type": "string",
+          "in": "path",
+          "x-ms-skip-url-encoding": true
+          
+        }
+      ]
+    },
+    "consumes": [
+      "application/json"
+    ],
+    "produces": [
+      "application/json"
+    ],
+    "paths": {
+      "/settings/{setting-name}": {
+        "patch": {
+          "tags": [
+            "Settings"
+          ],
+          "operationId": "UpdateSettings",
+          "summary": "Updates key vault account setting, stores it, then returns the setting name and value to the client.",
+          "description": "If false, all key management operations must happen through the Managed HSM data plane. If true, allows users to manage keys via the Azure Resource Manager.",
+          "parameters": [
+            {
+              "name": "setting-name",
+              "in": "path",
+              "required": true,
+              "type": "string",
+              "description": "The name of the account setting. Must be a valid settings option."
+            },
+            {
+              "name": "parameters",
+              "in": "body",
+              "required": true,
+              "x-ms-client-flatten": true,
+              "schema": {
+                "$ref": "#/definitions/UpdateSettingsRequest"
```

Comment:
After reviewing models, I don't understand how this is used. It's called "UpdateSetting" but apparently you don't update a setting - just enable or disable it. So at least now I see what your description refers to but I'm now confused on how a setting value is even changed. Is that strictly a management plane operation? If so, why is data plane enabling or disabling values? It's a very weird mix.

Reply:
So the value here is a boolean because the only settings suppored are boolean? But that also doesn't seem like enabling/disabling. If simple enabling/disabling a setting is what you're doing here, then use `enabled`. If the only supports values are boolean, 1) don't mention "enabling" or "disabling" the setting because that's not what this is doing, it's setting a value that happens to be a boolean, and 2) see my response below about this API supporting other types in the future. It has to remain compatible initially, so some sort of `type` property is necessary to describe the type like AppConfiguration did. At the very least, client libraries can parse/emit any intrinsic type as a string, but this also may not be ideal for code generation.